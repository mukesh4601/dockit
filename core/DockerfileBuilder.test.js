
let DockerfileBuilder = require('./DockerfileBuilder').DockerfileBuilder();

let dockerfileBuilder,runCommand,copyCommand,entrypointCommand;

beforeEach(()=>{
    dockerfileBuilder = new DockerfileBuilder();
    runCommand = new DockerfileBuilder.RunCommand(["/bin/bash","echo","${SAIF}"]);
    entrypointCommand = new DockerfileBuilder.EntryPointCommand(["/bin/bash"]);
    copyCommand = new DockerfileBuilder.CopyCommand("./","/");
});

afterEach(() => {
    dockerfileBuilder = null;
});

test('Test DockerfileBuilder Init',()=>{
    new DockerfileBuilder();
});

test('Test builder method returns builder instance',()=>{

    expect(dockerfileBuilder.withBaseImage('ubuntu') instanceof DockerfileBuilder ).toBeTruthy();
    expect(dockerfileBuilder.andRun(runCommand) instanceof DockerfileBuilder ).toBeTruthy();
    expect(dockerfileBuilder.andCopy(copyCommand) instanceof DockerfileBuilder ).toBeTruthy();
    expect(dockerfileBuilder.withEntryPointBuilder(entrypointCommand) instanceof DockerfileBuilder ).toBeTruthy();
    expect(dockerfileBuilder.withEnvs() instanceof DockerfileBuilder ).toBeTruthy();

});

test('Test DockerfileBuilder methods',()=>{
    expect(()=>{dockerfileBuilder.withBaseImage();}).toThrow();
    expect(dockerfileBuilder.withBaseImage('ubuntu')).not.toBeNull();
});

it("Create a Dockerfile",()=>{
    dockerfileBuilder.withBaseImage("ubuntu");
    dockerfileBuilder.andCopy(copyCommand);
    dockerfileBuilder.andCopy(copyCommand);
    dockerfileBuilder.andCopy(copyCommand);
    dockerfileBuilder.andRun(runCommand);
    dockerfileBuilder.andRun(runCommand);
    dockerfileBuilder.andRun(runCommand);
    dockerfileBuilder.withEntryPointBuilder(entrypointCommand);
    expect(dockerfileBuilder.toDockerfileString()).toMatchSnapshot();
});