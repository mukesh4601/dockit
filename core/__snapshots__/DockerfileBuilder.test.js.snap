// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a Dockerfile 1`] = `
"FROM ubuntu

COPY ./ /
COPY ./ /
COPY ./ /

RUN /bin/bash echo \${SAIF}
RUN /bin/bash echo \${SAIF}
RUN /bin/bash echo \${SAIF}
ENTRYPOINT /bin/bash"
`;
